tests/Installer/PluginInstaller.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Cake\Test\Composer\Installer;
+
+use Cake\Composer\Installer\PluginInstaller as PluginInstallerSrc;
+
+/**
+ * Test double for static methods in PluginInstaller
+ */
+class PluginInstaller extends PluginInstallerSrc
+{
+
+    /**
+     * Overriden to return a test-config file
+     *
+     * @param string $root
+     * @return string path to test plugins config file
+     */
+    public static function configFile($root)
+    {
+        $root = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'plugin-installer-test';
```

Comment:
This could be an issue if 2 apps are installing at the same time (ie, some deployment of multiple apps or some autoscaling script). Can you add some variant to it?